### ISSUE TITLE
False positive CS9223 in collection expression arguments

### ISSUE THREAD

**Issue by Spolutrean** [dotnet/roslyn#82562]
**Steps to Reproduce**:
Consider this code [.NET lab](https://lab.razor.fyi/#lZHNasJAEMexeNpjn2COm0ug18YERIooFQrNrZR23Qy6sNkNuxNtkLxH36CvWZJoNREP3eP_48fODPsZM_bi7MaJPJT-_ntcemU28Fp5wjycWa1RkrLGR-yWE87RoFMyYlIL72FVnc1JmsAjLJ5MmaMTa42TNGEHBgBQOLUThOBQZNboCp6Vpyb_oZWnqM2cimRd4_Q54Rzp7PMA4mN3aFyxLkH_pBTlWivZm5EHcOgqEIPBPQ8iqG-GC-FEfr0kS1t0Q1AnDmg7qzKYZhlPgf4KYSNQG63Z8Q6z06IveisetFrnNK_3EWUoAQkxvO0VbflD8N5NXbN6eSf95-hr9As):
```csharp
class MyCollection<T> : IEnumerable<T>
{
    private readonly List<T> _list;
    IEnumerator<T> IEnumerable<T>.GetEnumerator() => _list.GetEnumerator();
    IEnumerator IEnumerable.GetEnumerator() => _list.GetEnumerator();
    public MyCollection() { _list = new(); }
    public MyCollection(params MyCollection<T> other) { _list = new(other); }
    public void Add(T t) { _list.Add(t); }
}

class C
{
    public void M()
    {
        MyCollection<int> c = [with(1)]; // Creation of params collection 'MyCollection<int>' results in an infinite chain of invocation of constructor 'MyCollection<T>.MyCollection()'.
    }
}
```

**Diagnostic Id**: CS9223

**Expected Behavior**:
No CS9223 is reported, because the call stack will be `MyCollection(params MyCollection<T> other)` --> `MyCollection()` --> `.Add()` and no infinite recursion is happened. This exact behavior is fixed under a test  `CollectionExpressionTests_WithElement_Extra.ParamsCycle_MultipleConstructors`

**Actual Behavior**:
CS9223 is reported
